File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        int ans=n;
        for(int i=n; i<=100;i++)
        {
            int k=i, s=1;
            while(k>0)
            {
                int d=k%10;
                s*=d;
                k/=10;
            }
            if(s%t==0){
                ans=i;
                break;
            } 
        }
        return ans;
    }
}

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        
        for(int i=n;i<=100;i++)
        {
            int x=i, p=1;
            while(x>0)
            {
                int d=x%10;
                p*=d;
                x/=10;
            }
            if(p%t == 0)
            {
                
                return i;
            }
        }
        return n;
    }
};